airflow/providers/amazon/aws/sensors/opensearch_serverless.py

```diff
@@ -0,0 +1,110 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any, Sequence
+
+from airflow.exceptions import AirflowException, AirflowSkipException
+from airflow.providers.amazon.aws.hooks.opensearch_serverless import OpenSearchServerlessHook
+from airflow.providers.amazon.aws.sensors.base_aws import AwsBaseSensor
+from airflow.providers.amazon.aws.utils.mixins import aws_template_fields
+from airflow.utils.helpers import exactly_one
+
+if TYPE_CHECKING:
+    from airflow.utils.context import Context
+
+
+class OpenSearchServerlessCollectionActiveSensor(AwsBaseSensor[OpenSearchServerlessHook]):
+    """
+    Poll the state of the Collection until it reaches a terminal state; fails if the query fails.
+
+    .. seealso::
+        For more information on how to use this sensor, take a look at the guide:
+        :ref:`howto/sensor:OpenSearchServerlessCollectionAvailableSensor`
+
+    :param collection_id: A collection ID. You can't provide a name and an ID in the same request.
+    :param collection_name: A collection name. You can't provide a name and an ID in the same request.
+
+    :param deferrable: If True, the sensor will operate in deferrable more. This mode requires aiobotocore
+        module to be installed.
+        (default: False, but can be overridden in config file by setting default_deferrable to True)
```

Comment:
I could not find this parameter in this class neither in its parent classes. And you dont seem to use it in the implementation. Left over I guess.

Reply:
Ah, no, I made the trigger for that and forgot to implement it there.  Good catch, I'll add it now.

Reply:
Implemented.